prow/github/client.go

```diff
@@ -1717,6 +1718,40 @@ func (c *client) GetSingleCommit(org, repo, SHA string) (SingleCommit, error) {
 	return commit, err
 }
 
+// List commits on a repository
+// If path is empty, then it is not used as parameter in the URL.
+// See https://developer.github.com/v3/repos/commits/#list-commits-on-a-repository
+func (c *client) ListCommits(org, repo, sha, path, author, since, until string) ([]SingleCommit, error) {
+	c.log("ListCommits", org, repo, path)
+	url := fmt.Sprintf("/repos/%s/%s/commits", org, repo)
+	var args []string
+	if sha != "" {
+		args = append(args, fmt.Sprintf("sha=%s", sha))
+	}
+	if path != "" {
+		args = append(args, fmt.Sprintf("path=%s", path))
+	}
+	if author != "" {
+		args = append(args, fmt.Sprintf("author=%s", author))
+	}
+	if since != "" {
+		args = append(args, fmt.Sprintf("since=%s", since))
+	}
+	if until != "" {
+		args = append(args, fmt.Sprintf("until=%s", until))
```

Comment:
I don't believe we should construct URL queries like that, there's some encoding involved... Looks like `net/url` has a struct for that: https://golang.org/pkg/net/url/#Values

Reply:
@petr-muller 

Thanks for reviewing ... at such hours for yours.
Actually, the idea of using this API is rejected by @stevekuznetsov 

I will use
```
v := url.Values{}
v.Set("name", "Ava")
v.Encode()
```
when we decide to implement this in the future.